cpp/build/Makefile

```diff
@@ -23,6 +23,9 @@ top_srcdir := $(abspath $(dir $(lastword $(MAKEFILE_LIST)))../../)
 
 include $(top_srcdir)/cpp/build/support.mk
 
+# Used to detect SmartOS (variable will be "joyent")
+UNAME_V := $(shell uname -v 2>/dev/null | sed -e 's/_.*//')
```

Comment:
Why can't you use UNAME instead?

Reply:
The issue here is that `UNAME` on SmartOS overlaps with other operating systems.

```
$ uname 
SunOS
$ uname -v
joyent_20180510T153535Z
```

This will overlap with Solaris, as well as other [illumos](http://illumos.org/) based operating systems.  The issue here is that using `/opt/local` as a prefix is something that is fairly specific to SmartOS.  Just detecting `UNAME == SunOS` would work for the SmartOS case, but would ultimately fail for Solaris or other Illumos based operating systems if support for them were to ever be considered in the future.

Reply:
I'm testing a patch right now that won't hardcode `/opt/local` and will also not rely on `uname -v` - it should just work with UNAME instead.  I'll know more next week when I can test it better.